be/src/vec/sink/vtablet_sink.cpp

```diff
@@ -472,6 +473,35 @@ Status VNodeChannel::open_wait() {
     return status;
 }
 
+void VNodeChannel::open_partition(int64_t partition_id) {
+    SCOPED_CONSUME_MEM_TRACKER(_node_channel_tracker.get());
+    PartitionOpenRequest request;
+    auto load_id = std::make_shared<PUniqueId>(_parent->_load_id);
+    request.set_allocated_id(load_id.get());
+    request.set_index_id(_index_channel->_index_id);
+    for (auto& tablet : _all_tablets) {
+        if (partition_id == tablet.partition_id) {
+            auto ptablet = request.add_tablets();
+            ptablet->set_partition_id(tablet.partition_id);
+            ptablet->set_tablet_id(tablet.tablet_id);
+        }
+    }
+
+    PartitionOpenClosure<PartitionOpenResult>* partition_open_closure =
+            new PartitionOpenClosure<PartitionOpenResult>(this);
+    partition_open_closure->ref();
+
+    // This ref is for RPC's reference
+    partition_open_closure->ref();
+    partition_open_closure->cntl.set_timeout_ms(config::partition_open_rpc_timeout_sec * 1000);
```

Comment:
We should use remaining of the task timeout to avoid a long running task failed due to a rpc error.